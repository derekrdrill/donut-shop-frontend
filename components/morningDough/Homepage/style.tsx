import styled from 'styled-components';
import { Grid } from '@mui/material';

export const HomepageRoot = styled(Grid)({
  overflowX: 'hidden',
});

export const HomepageImage = styled.img<{
  $isOnRightSide?: boolean;
  $isOnLeftSide?: boolean;
}>(({ $isOnLeftSide, $isOnRightSide }) => ({
  height: 300,
  width: 300,
  transform: $isOnLeftSide ? 'translateX(-100px)' : $isOnRightSide ? 'translate(100px)' : 'none',
}));

export const HomepageText = styled.h1<{
  $fontColor?: string;
  $fontSize: number;
  $isSubtext?: boolean;
}>(({ $fontColor, $fontSize, $isSubtext }) => ({
  color: $fontColor,
  fontFamily: $isSubtext ? `'Josefin Sans', sans-serif` : `'Lilita One', cursive !important`,
  fontSize: $fontSize,
  margin: '15px 0',
  textAlign: 'center',
  textTransform: 'uppercase',
}));

export const HomePageTextSpan = styled.span<{ $fontColor: string }>(({ $fontColor }) => ({
  color: $fontColor,
}));
