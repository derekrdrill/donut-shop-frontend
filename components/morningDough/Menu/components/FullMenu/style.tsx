import { Grid, Typography } from '@mui/material';
import styled from 'styled-components';

export const FullMenuHeadContainer = styled(Grid)({
  backgroundColor: '#fef3f1',
  minHeight: 250,
  opacity: 0.8,
});

export const FullMenuHeadTextItem = styled(Grid)({
  padding: 20,
});

export const FullMenuContainer = styled(Grid)({
  paddingBottom: 40,
  margin: '40px 4% 0px 4%',
});

export const FullMenuText = styled(Typography)<{
  $fontColor?: string;
  $isSubtext?: boolean;
}>(({ $fontColor, $isSubtext }) => [
  {
    color: $fontColor,
    fontFamily: `'Lilita One', cursive !important`,
    margin: '15px 0',
  },
  $isSubtext && {
    fontFamily: `'Rubik', sans-serif !important`,
  },
]);

export const FullMenuHeadImage = styled.img({
  height: 225,
  width: 225,
});

export const FullMenuItemRoot = styled(Grid)({
  height: 200,
  padding: 10,
});

export const FullMenuItemContainer = styled(Grid)({
  ':hover': {
    backgroundColor: '#f8d7ff',
  },
  border: '1px solid #ababab',
  borderRadius: 10,
  cursor: 'pointer',
  maxWidth: 340,
  transition: 'all 250ms ease-in',
});

export const FullMenuItemImage = styled.img({
  height: 150,
  width: 150,
});
