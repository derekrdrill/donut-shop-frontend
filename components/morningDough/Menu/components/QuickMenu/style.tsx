import styled from 'styled-components';
import { Grid } from '@mui/material';

export const QuickMenuImage = styled.img({
  height: 100,
  transition: 'all 50ms ease-in',
  width: 100,
});

export const QuickMenuTabContainer = styled(Grid)<{ $isMenuTabActive?: boolean }>(
  ({ $isMenuTabActive }) => ({
    '&:hover': {
      backgroundColor: '#bb00e0',
      color: '#FFFFFF',
      opacity: !$isMenuTabActive ? 0.6 : 1,
    },
    backgroundColor: $isMenuTabActive ? '#bb00e0' : 'inherit',
    color: $isMenuTabActive ? '#FFFFFF' : 'inherit',
    cursor: 'pointer',
    opacity: 0.8,
    padding: 8,
  }),
);

export const QuickMenuItemContainer = styled(Grid)({
  '&:hover': {
    img: {
      opacity: 0.8,
    },
    '.MuiTypography-root': {
      textDecoration: 'underline',
    },
    border: '1px gainsboro solid',
    borderRadius: 20,
    boxShadow: 'rgba(0, 0, 0, 0.16) 0px 3px 6px, rgba(0, 0, 0, 0.23) 0px 3px 6px',
  },
  cursor: 'pointer',
  maxHeight: 185,
  minWidth: 100,
  padding: 15,
});

export const QuickViewMenuItemsRoot = styled(Grid)({
  backgroundColor: '#f5f5f5',
  height: 354,
  overflowY: 'auto',
  padding: 20,
});
