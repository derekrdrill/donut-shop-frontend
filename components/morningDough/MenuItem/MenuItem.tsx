import * as React from 'react';
import { useSearchParams } from 'next/navigation';
import { useRouter } from 'next/router';
import { Button, Grid } from '@mui/material';
import KeyboardDoubleArrowLeftIcon from '@mui/icons-material/KeyboardDoubleArrowLeft';

import GlobalContext from '../../../context/GlobalContext';

import MenuItemCounter from './components/MenuItemCounter';
import MenuItemSizeToggle from './components/MenuItemSizeToggle';
import MenuItemFlavorSelect from './components/MenuItemFlavorSelect';
import MenuItemDairySelect from './components/MenuItemDairySelect';
import MenuItemIceSelect from './components/MenuItemIceSelect';
import MenuItemSugarSelect from './components/MenuItemSugarSelect';
import MenuItemCreamCheeseSelect from './components/MenuItemCreamCheeseSelect';
import MenuItemButterSelect from './components/MenuItemButterSelect';

import { setMyBagHelper, updateCurrentMyBag } from './actions/MenuItemActions';
import { setAlertItem } from '../../../components/common/Alert/actions/AlertActions';

import {
  MenuItemImage,
  MenuItemDetailContainer,
  MenuItemOrderButton,
  MenuItemDetailText,
} from './style';

import { FullMenuItem } from '../Menu/assets/data/FULL_MENU';

export const getSugarString = (sugar: boolean) => (sugar ? 'yes' : 'no');
export const getSubmitButtonString = (orderID: string | null) =>
  orderID ? 'Update My Bag' : 'Add to My Bag';
interface MenuItemProps {
  fullMenu: FullMenuItem[];
}

const MenuItem = ({ fullMenu }: MenuItemProps) => {
  const router = useRouter();
  const searchParams = useSearchParams();
  const menuItemID = searchParams.get('menuItem');
  const orderID = searchParams.get('orderID');

  const {
    state: { myBag },
    dispatch,
  } = React.useContext(GlobalContext);

  const [menuItemData, setMenuItemData] = React.useState<FullMenuItem>(
    fullMenu.filter(menuItem => menuItem.key === menuItemID)[0],
  );

  const [selectedCount, setSelectedCount] = React.useState<number>(1);
  const [selectedDairy, setSelectedDairy] = React.useState<string>('');
  const [selectedFlavor, setSelectedFlavor] = React.useState<string>('');
  const [selectedIce, setSelectedIce] = React.useState<string>('normal');
  const [selectedSize, setSelectedSize] = React.useState<string>('md');
  const [selectedSugar, setSelectedSugar] = React.useState<string>('');
  const [selectedCreamCheese, setSelectedCreamCheese] = React.useState<string>('');
  const [selectedButter, setSelectedButter] = React.useState<string>('');

  const handleChange = (event: React.MouseEvent<HTMLElement>, newSelectedSize: string) => {
    setSelectedSize(newSelectedSize);
  };

  React.useEffect(() => {
    setMenuItemData(fullMenu.filter(menuItem => menuItem.key === menuItemID)[0]);
    setSelectedCount(1);
    setSelectedDairy('');
    setSelectedFlavor('');
    setSelectedIce('normal');
    setSelectedSize('md');
    setSelectedSugar('');
    setSelectedCreamCheese('');
    setSelectedButter('');
  }, [menuItemID]);

  React.useEffect(() => {
    if (orderID && myBag && myBag.length > 0) {
      const { butter, creamCheese, dairy, flavor, ice, quantity, size, sugar } = myBag.filter(
        myBagItem => myBagItem.orderID === orderID,
      )[0];

      setSelectedCount(quantity);
      setSelectedDairy(dairy ?? '');
      setSelectedFlavor(flavor ?? '');
      setSelectedIce(ice ?? '');
      setSelectedSize(size);
      setSelectedSugar(getSugarString(sugar));
      setSelectedCreamCheese(creamCheese ?? '');
      setSelectedButter(butter ?? '');
    }
  }, [orderID]);

  return (
    <Grid container>
      <Grid item xs={12} lg={7}>
        <MenuItemDetailContainer container columnSpacing={2} rowSpacing={2}>
          <Grid item xs={12}>
            <Button
              color='secondary'
              onClick={
                /* istanbul ignore next */
                // () => navigate('/menu')
                () => router.push('/menu')
              }
              startIcon={<KeyboardDoubleArrowLeftIcon />}
              variant='text'
            >
              Back to full menu
            </Button>
          </Grid>
          <Grid
            item
            xs={12}
            sm={6}
            lg={12}
            sx={{ borderRight: { xs: 'none', sm: 'solid 1px gainsboro', lg: 'none' } }}
          >
            <Grid item xs={12} display={{ lg: 'none' }}>
              <MenuItemImage height={200} src={menuItemData?.image} width={200} />
            </Grid>
            <Grid item xs={12}>
              <MenuItemDetailText color='#272020' variant='h2'>
                {menuItemData?.name}
              </MenuItemDetailText>
            </Grid>
          </Grid>
          <Grid item xs={12} sm={6} lg={12}>
            <Grid container spacing={2}>
              <MenuItemSizeToggle
                menuItemData={menuItemData}
                selectedSize={selectedSize}
                setSelectedSize={setSelectedSize}
              />
              <MenuItemDairySelect
                menuItemData={menuItemData}
                selectedDairy={selectedDairy}
                setSelectedDairy={setSelectedDairy}
              />
              <MenuItemFlavorSelect
                menuItemData={menuItemData}
                selectedFlavor={selectedFlavor}
                setSelectedFlavor={setSelectedFlavor}
              />
              <MenuItemIceSelect
                menuItemData={menuItemData}
                selectedIce={selectedIce}
                setSelectedIce={setSelectedIce}
              />
              <MenuItemSugarSelect
                menuItemData={menuItemData}
                selectedSugar={selectedSugar}
                setSelectedSugar={setSelectedSugar}
              />
              <MenuItemCreamCheeseSelect
                menuItemData={menuItemData}
                selectedCreamCheese={selectedCreamCheese}
                setSelectedCreamCheese={setSelectedCreamCheese}
              />
              <MenuItemButterSelect
                menuItemData={menuItemData}
                selectedButter={selectedButter}
                setSelectedButter={setSelectedButter}
              />
              <Grid item xs={12}>
                <Grid container>
                  <Grid item xs={12} md={6} lg={4}>
                    <MenuItemCounter count={selectedCount} setCount={setSelectedCount} />
                  </Grid>
                </Grid>
              </Grid>
              <Grid item xs={12} lg={4}>
                <MenuItemOrderButton
                  buttonColor='#f2755f'
                  buttonHoverColor='#f59180'
                  fullWidth
                  marginTop={50}
                  onClick={
                    /* istanbul ignore next */
                    async () => {
                      if (orderID) {
                        dispatch(
                          await updateCurrentMyBag(
                            selectedCount,
                            selectedDairy,
                            selectedFlavor,
                            selectedSugar,
                            selectedSize,
                            selectedIce,
                            selectedCreamCheese,
                            selectedButter,
                            myBag,
                            orderID,
                            menuItemData?.price,
                          ),
                        );

                        dispatch(
                          await setAlertItem(
                            `${menuItemData?.name} item has been updated`,
                            true,
                            'success',
                          ),
                        );
                      } else {
                        dispatch(
                          await setMyBagHelper(
                            menuItemID,
                            selectedCount,
                            selectedDairy,
                            selectedFlavor,
                            selectedSugar,
                            selectedSize,
                            selectedIce,
                            selectedCreamCheese,
                            selectedButter,
                            myBag,
                            menuItemData?.name,
                            menuItemData?.category,
                            menuItemData.bottled ?? false,
                            menuItemData.soda ?? false,
                            menuItemData?.subCategory,
                            menuItemData?.image,
                            `${new Date().getTime()}-${menuItemID}`,
                            menuItemData?.price,
                          ),
                        );

                        dispatch(
                          await setAlertItem(
                            `${selectedCount} ${menuItemData?.name}${
                              selectedCount > 1 ? 's' : ''
                            } added to bag`,
                            true,
                            'success',
                          ),
                        );
                      }
                    }
                  }
                  size='large'
                  variant='contained'
                >
                  {getSubmitButtonString(orderID)}
                </MenuItemOrderButton>
              </Grid>
            </Grid>
          </Grid>
        </MenuItemDetailContainer>
      </Grid>
      <Grid item xs={5} display={{ xs: 'none', lg: 'block' }}>
        <MenuItemImage height={500} src={menuItemData?.image} width={500} />
      </Grid>
    </Grid>
  );
};

export default MenuItem;
