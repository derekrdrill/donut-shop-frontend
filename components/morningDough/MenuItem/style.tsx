import styled from 'styled-components';
import { Button, Grid, ToggleButton, Typography } from '@mui/material';
import ActionButton from '../../common/ActionButton/ActionButton';

export const MenuItemImage = styled.img<{ height: number; width: number }>(({ height, width }) => ({
  height: height,
  widht: width,
}));

export const MenuItemDetailContainer = styled(Grid)({
  minHeight: 500,
  padding: '40px 60px',
});

export const MenuItemDetailText = styled(Typography)<{
  $fontColor?: string;
  $isSubtext?: boolean;
}>(({ $fontColor, $isSubtext }) => [
  {
    color: $fontColor,
    fontFamily: `'Lilita One', cursive !important`,
    margin: '15px 0',
  },
  $isSubtext && {
    fontFamily: `'Rubik', sans-serif !important`,
  },
]);

export const MenuItemToggleButton = styled(ToggleButton)({
  width: 75,
});

export const MenuItemOrderButton = styled(ActionButton)({
  borderRadius: '20px !important',
});
