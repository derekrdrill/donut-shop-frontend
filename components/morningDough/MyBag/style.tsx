import styled from 'styled-components';
import { Badge, Grid } from '@mui/material';
import ShoppingBagIcon from '@mui/icons-material/ShoppingBag';

export const MyBagRoot = styled(Grid)({
  marginTop: 40,
});

export const MyBagIconSVG = styled(ShoppingBagIcon)({
  marginTop: 7,
  transition: 'all 250ms linear',
  transform: 'scale(1.4)',
});

export const MyBagIconBadge = styled(Badge)({
  '.MuiBadge-badge': {
    backgroundColor: 'white',
    color: '#6a017f',
    border: '1px #6a017f solid',
    transition: 'all 250ms linear',
  },
});

export const MyBagCard = styled(Grid)({
  border: '1px solid gainsboro',
  borderRadius: 5,
  margin: '0px 10px !important',
  padding: 20,
});

export const MyBagItemContainer = styled(Grid)({
  height: 390,
  overflowY: 'auto',
});

export const MyBagItemRow = styled(Grid)({
  borderBottom: '1px solid gainsboro',
  borderTop: '1px solid gainsboro',
  padding: 15,
});

export const MyBagItemImage = styled.img({
  height: 40,
  width: 40,
});

export const MyBagIconContainer = styled(Grid)({
  '&:hover': {
    '.MuiSvgIcon-root': {
      color: '#a501c6',
      transform: 'scale(1.6)',
    },
    '.MuiBadge-root': {
      transform: 'scale(1.2)',
      zIndex: 4,
    },
  },
  cursor: 'pointer',
});

export const MyBagPaymentContainer = styled(Grid)({
  borderTop: '1px solid gainsboro',
  padding: 15,
});
