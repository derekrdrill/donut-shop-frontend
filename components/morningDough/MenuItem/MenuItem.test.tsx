jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
  useParams: () => ({
    menuItemID: 'chocolateDonut',
    orderID: 'chocolateDonut-3218737128',
  }),
}));

import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import GlobalContext from '../../context/GlobalContext';

import MenuItem, { getSugarString, getSubmitButtonString } from './MenuItem';

import { MenuItemDetailText } from './style';

describe('MenuItem unit tests', () => {
  const myBag = [
    {
      bottled: false,
      butter: null,
      category: 'donuts',
      creamCheese: null,
      dairy: null,
      flavor: null,
      ice: 'normal',
      menuItemID: 'chocolateDonut',
      menuItemImage: 'http://localhost:3000/chocolateDonut.png',
      name: 'Chocolate',
      orderID: 'chocolateDonut-3218737128',
      quantity: 5,
      size: 'md',
      soda: false,
      subCategory: 'donuts',
      sugar: false,
    },
    {
      bottled: false,
      butter: null,
      category: 'drink',
      creamCheese: null,
      dairy: 'cream',
      flavor: 'frenchVanilla',
      ice: 'normal',
      menuItemID: 'hotCoffee',
      menuItemImage: 'http://localhost:3000/hotCoffee.png',
      name: 'Hot Coffee',
      orderID: '1681765872483-menuItemID',
      quantity: 3,
      size: 'md',
      soda: false,
      subCategory: 'hotDrinks',
      sugar: true,
    },
  ];

  it('renders as expected', () => {
    const result = renderer
      .create(
        <GlobalContext.Provider value={{ dispatch: jest.fn(), state: { myBag } }}>
          <MenuItem
            fullMenu={[
              {
                key: 'chocolateDonut',
                name: 'Chocolate',
                image: 'chocolateDonut',
                category: 'donuts',
                subCategory: 'donuts',
              },
            ]}
          />
          ,
        </GlobalContext.Provider>,
      )
      .toJSON();
    expect(result).toMatchSnapshot();
  });

  it('runs getSugarString', () => {
    expect(getSugarString(true)).toEqual('yes');
    expect(getSugarString(false)).toEqual('no');
  });

  it('runs getSubmitButtonString', () => {
    expect(getSubmitButtonString('1234')).toEqual('Update My Bag');
    expect(getSubmitButtonString(null)).toEqual('Add to My Bag');
  });

  it('renders MenuItemDetailText correctly (isSubText)', async () => {
    render(<MenuItemDetailText data-testid='isSubtext' $isSubtext />);

    const menuItemDetailText = await screen.findAllByTestId('isSubtext');

    expect(menuItemDetailText[0]).toHaveClass('sc-hLseeU hefLba');
  });

  it('renders MenuItemDetailText correctly (fontColor)', async () => {
    render(<MenuItemDetailText data-testid='fontColor' $fontColor='red' />);

    const menuItemDetailText = await screen.findAllByTestId('fontColor');

    expect(menuItemDetailText[0]).toHaveClass('sc-hLseeU jQZkzZ');
  });
});
