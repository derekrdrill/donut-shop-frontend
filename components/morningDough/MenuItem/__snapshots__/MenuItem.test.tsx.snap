// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem unit tests renders as expected 1`] = `
[
  <div
    className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-7 css-117e5y7-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container sc-dmqHEX bmnJWn css-mhc70k-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium css-1d3f8j8-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardDoubleArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.59 18 19 16.59 14.42 12 19 7.41 17.59 6l-6 6z"
                />
                <path
                  d="m11 18 1.41-1.41L7.83 12l4.58-4.59L11 6l-6 6z"
                />
              </svg>
            </span>
            Back to full menu
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-12 css-1byk9tm-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-60n250-MuiGrid-root"
          >
            <img
              className="sc-gueYoa zpOOU"
              height={200}
              src="chocolateDonut"
              width={200}
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h2 sc-hLseeU kfctxS css-zxcin0-MuiTypography-root"
            >
              Chocolate
            </h2>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-12 css-ose4ye-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
              >
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-1e6jbaj-MuiGrid-root"
                >
                  <div
                    className="MuiGrid-root MuiGrid-container css-1sjizby-MuiGrid-root"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                    >
                      <label
                        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                      >
                        How many?
                      </label>
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                    >
                      <button
                        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth Mui-disabled MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth sc-beqWaB fyBusN css-xsqe0f-MuiButtonBase-root-MuiButton-root"
                        disabled={true}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onContextMenu={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={-1}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="RemoveIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 13H5v-2h14v2z"
                          />
                        </svg>
                      </button>
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                    >
                      <h6
                        className="MuiTypography-root MuiTypography-h6 css-1kgxqm0-MuiTypography-root"
                      >
                        1
                      </h6>
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                    >
                      <div
                        className="MuiGrid-root MuiGrid-container css-16g5rpk-MuiGrid-root"
                      >
                        <button
                          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth sc-beqWaB jKMOlt css-xsqe0f-MuiButtonBase-root-MuiButton-root"
                          disabled={false}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          type="button"
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="AddIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                            />
                          </svg>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-4 css-1qm8set-MuiGrid-root"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth sc-beqWaB fKdhAl css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                Update My Bag
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-bo8z12-MuiGrid-root"
    >
      <img
        className="sc-gueYoa itNSnk"
        height={500}
        src="chocolateDonut"
        width={500}
      />
    </div>
  </div>,
  ",",
]
`;
